openstack/elb/v3/l7policies/requests.go

```diff
@@ -62,11 +62,73 @@ type CreateOpts struct {
 	// Only valid if action is REDIRECT_TO_LISTENER.
 	RedirectListenerID string `json:"redirect_listener_id,omitempty"`
 
+	// Requests matching this policy will be redirected to the URL.
+	// Only valid if action is REDIRECT_TO_URL.
+	RedirectUrlConfig *RedirectUrlConfig `json:"redirect_url_config,omitempty"`
+
+	// Requests matching this policy will be redirected to the configuration of the page.
+	// Only valid if action is FIXED_RESPONSE.
+	FixedResponseConfig *FixedResponseConfig `json:"fixed_response_config,omitempty"`
+
+	// The config of the redirected pool.
+	// Only valid if action is REDIRECT_TO_POOL.
+	RedirectPoolsExtendConfig *RedirectPoolsExtendConfig `json:"redirect_pools_extend_config,omitempty"`
+
 	// The administrative state of the Loadbalancer. A valid value is true (UP)
 	// or false (DOWN).
 	AdminStateUp *bool `json:"admin_state_up,omitempty"`
 }
 
+type RedirectUrlConfig struct {
+	// The protocol for redirection.
+	Protocol string `json:"protocol,omitempty"`
+
+	// The host name that requests are redirected to.
+	Host string `json:"host,omitempty"`
+
+	// The port that requests are redirected to.
+	Port string `json:"port,omitempty"`
+
+	// The path that requests are redirected to.
+	Path string `json:"path,omitempty"`
+
+	// The query string set in the URL for redirection.
+	Query string `json:"query"`
```

Comment:
missing `omitempty`? and check the other places

Reply:
no, it can be empty